packages/hardhat-core/src/builtin-tasks/utils/solidity-files-cache.ts

```diff
@@ -4,7 +4,7 @@ import { LoDashStatic } from "lodash";
 import * as path from "path";
 
 import { SOLIDITY_FILES_CACHE_FILENAME } from "../../internal/constants";
-import type { ProjectPaths, SolcConfig } from "../../types";
+import type { ProjectPathsConfig, SolcConfig } from "../../types";
```

Comment:
Not very important, but the `io-ts` type wasn't renamed to `ProjectPathsConfig`.

Reply:
Yes, I noticed that all of the `io-ts` config types have outdated names. I didn't update all of them here to keep this PR small.

Reply:
I added a TODO